### DISCUSSION TITLE
Opentelemetry Jaeger Exporter sends only few traces to Jager

### DISCUSSION THREAD

**Discussion by SimoneGiusso** [open-telemetry/opentelemetry-java-instrumentation#6256] [Q&A] [answered]
Hi,

I was testing this `opentelemetry-exporter-jaeger` with a demo application using Jaeger as Collector. This demo application can be found in my [repo](https://github.com/SimoneGiusso/OpenTelemetryJaegerDemo) and it consists of a simple controller composed by just one endpoint to get a random number.

After I had run this application locally I tried to call the endpoint (with Postman). The application was run together with the All-in-One executable Jaeger service (by means of docker, see official [documentation](https://www.jaegertracing.io/docs/1.35/getting-started/#all-in-one)).

I called the endpoint of my demo application more the 10 times in around 5 seconds:

<img width="1324" alt="Screenshot 2022-07-03 at 18 07 00" src="https://user-images.githubusercontent.com/26305592/177047856-e2c3f1df-c92d-4954-ba2e-233e876f4761.png">

However from the Jaeger UI only one call was traced (the third one):

<img width="1324" alt="Screenshot 2022-07-03 at 18 07 00" src="https://user-images.githubusercontent.com/26305592/177047902-96eba512-6d07-4b09-bb50-db673197666f.png">

It looks like some calls are randomly traced (actually only a few of them). Is there anyone know if it's a configuration problem or something else?

Thanks!

**Comment by mateuszrzeszutek** [comment]
Hey @SimoneGiusso ,
Have you perhaps set up a sampler? Can you share the configuration that you're using to run the app? (Env vars; system properties, the whole command line if possible; dockerfile if you're using it)

**Reply by SimoneGiusso** [reply]
Hi @mateuszrzeszutek thanks for your reply.

Actually I'm simply running the app through my IDE (intelliJIDEA as I'm doing just some tests now) so it's not running in a docker container and I'm not passing any specific env vars and I'm not using a sampler. All is running locally (Jaeger container + Demo APP) on my M1 Air mac. When solved this problem I'll have other tests to do.

**Reply by mateuszrzeszutek** [reply]
Oh, I noticed that you're using Spring Cloud Sleuth; I was under the impression that you might be using the javaagent, thus my question about configuration; nevermind that then.

Try using the `opentelemetry-exporter-logging` and check what kind of spans are printed out to the console - let's verify that these spans are being generated properly in the first place.

**Reply by SimoneGiusso** [reply]
Hi. I added `opentelemetry-exporter-logging` (see 2nd commit in my repo). I kept the default [configuration](https://github.com/open-telemetry/opentelemetry-collector/blob/main/exporter/loggingexporter/README.md).

I can clearly see in the console that `LoggingSpanExporter` prints something when the trace is visible in the Jaeger GUI. From the [log](https://github.com/open-telemetry/opentelemetry-java-instrumentation/files/9049622/OpenTelemetryJaegerDemo_Log.txt) attached you can see that I called the endpoint 6 times. However the trace seems to be exported only twice.

**Reply by mateuszrzeszutek** [reply]
Hmm, that means that something is not exactly right with the application itself/the default Spring Sleuth configuration; we've excluded the possibility of jaeger or jaeger exporter losing things. Can you create an issue about this in the Spring Sleuth repo? There doesn't seem to be anything broken with OpenTelemetry itself, at the first glance.

**Reply by SimoneGiusso** [reply]
Thank a lot for your support. I raised an issue in the [spring-cloud-sleuth-otel](https://github.com/spring-projects-experimental/spring-cloud-sleuth-otel/issues/117) repo.

**Comment by SimoneGiusso** [comment, marked as answer]
Hi. Thanks @mateuszrzeszutek for your support. The problem was solved: a spring sleuth otel configuration was needed to sample all the traces: `spring.sleuth.otel.config.trace-id-ratio-based=1.0`
